internal/manifests/collector/service.go

```diff
@@ -106,6 +106,15 @@ func Service(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelemet
 		return nil
 	}
 
+	// set appProtocol to h2c for grpc ports on OpenShift.
+	// OpenShift uses HA proxy that uses appProtocol for its configuration.
+	for i, _ := range ports {
+		h2c := "h2c"
+		if otelcol.Spec.Ingress.Type == v1alpha1.IngressTypeRoute && ports[i].AppProtocol != nil && *(ports[i].AppProtocol) == "grpc" {
```

Comment:
do we need to do the pointer stuff here? would we be able to use something strings.Equal? or something like that?

Reply:
I have changed it to ` strings.EqualFold(*ports[i].AppProtocol, "grpc")`. The nil check seems to be needed here